src/servers/main.cc

```diff
@@ -1486,6 +1481,11 @@ main(int argc, char** argv)
     exit(1);
   }
 
+  // Start the HTTP, GRPC, and metrics endpoints.
+  if (!StartEndpoints(server, trace_manager, shm_manager)) {
+    exit(1);
+  }
+
```

Comment:
I moved the registration of endpoints after the registration of signal handling so that when the server is ready the destructors are called properly.